dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/mbeans/PluginGenerator.java

```diff
@@ -905,65 +887,29 @@ public static void copyFile(File in, File out)
-            synchronized (transformerLock) {
-
-                final String defaultTransformerFactory = getJVMProperty(TRANSFORMER_FACTORY_JVM_PROPERTY_NAME);
-
-                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                    Tr.debug(tc, "JDK = " + JavaInfo.vendor() + ", JDK level = " + JavaInfo.majorVersion() + "." + JavaInfo.minorVersion() + ", current TF jvm property value = "
-                                 + defaultTransformerFactory);
-                }
-
-                AccessController.doPrivileged(new PrivilegedAction<Object>() {
-                    @Override
-                    public Object run() {
-                        System.setProperty(TRANSFORMER_FACTORY_JVM_PROPERTY_NAME, "org.apache.xalan.processor.TransformerFactoryImpl");
-                        return null;
-                    }
-                });
-
-                tf = TransformerFactory.newInstance();
-
-                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                    Tr.debug(tc, "IBM JDK : Use transformer factory: " + tf.getClass().getName());
-                }
-
-                AccessController.doPrivileged(new PrivilegedAction<Object>() {
-                    @Override
-                    public Object run() {
-                        if (defaultTransformerFactory != null)
-                            System.setProperty(TRANSFORMER_FACTORY_JVM_PROPERTY_NAME, defaultTransformerFactory);
-                        else
-                            System.clearProperty(TRANSFORMER_FACTORY_JVM_PROPERTY_NAME);
-                        return null;
-                    }
-                });
-
-                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                    Tr.debug(tc, "IBM JDK : TF jvm property value restored: " + getJVMProperty(TRANSFORMER_FACTORY_JVM_PROPERTY_NAME));
-                }
-            }
-        } else {
-            tf = TransformerFactory.newInstance();
+        //We want to use the the Interpretive processor (Apache Xalan) but need to work out which one
+        //This is either going to be the one package with an IBM Java SDK or the one packaged as a fallback by Oracle
+        boolean changeTransformer = JavaInfo.isAvailable(IBM_XLTXEJ_COMPILED_TRANSFORMER_FACTORY_CLASS_NAME) &&
+                                    JavaInfo.isAvailable(XALAN_TRANSFORMER_FACTORY_CLASS_NAME) &&
+                                    JavaInfo.isAvailable(SAX_LEXICAL_HANDLER_CLASS_NAME);
+        
+        String transformerFactory = changeTransformer ? XALAN_TRANSFORMER_FACTORY_CLASS_NAME : FALLBACK_XALAN_TRANSFORMER_FACTORY_CLASS_NAME;
+        
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "JDK : Use transformer factory: " + transformerFactory);
+        }
+        
+        tf = TransformerFactory.newInstance(transformerFactory, ClassLoader.getSystemClassLoader());
```

Comment:
I think this has the potential to break in future if the FALLBACK_XALAN processor not present in any JVM which doesn't have all of XALAN, IBM_XLTXEJ the SAX lexical handler.

I think it would be better to retain the old behaviour and call `TransformerFactory.newInstance()` to obtain the default implementation if `changeTransformer` is false.